### DISCUSSION TITLE
Optional Pingback / Telemetry functionality

### DISCUSSION THREAD

**Discussion by ArneTR** [green-coding-solutions/green-metrics-tool#582] [Ideas]
I wanted to open this issue to users and the community as for the core developer team the wish was raised to be more aware of how many people are using the Green Metrics Tool.

The motivation behind it was to increase motivation and the feeling of impact when developing the tool.

Currently, by design, the tool is fully autonomous and does not do any telemetry. It is not needed for it's use case and also not needed for security or any other case.

However, for aforementioned reasons, the core team is thinking of introducing an optional "ping" whenever the install script is run.
The ping would transmit the commit hash of the locally installed repository version and, due to the technical nature, the time when the install was done.

Also we are aware that the IP is transmitted due to technical reasons. We would not log it, but maybe reserve the opportunity to extract the country from it and use that information.

The technical implementation would be a final step in the install script where the user is greeted with a message like this:

> Do you want to send a ping to the developers of the tool let them know that the tool was used? This ping is optional and you can of course press no. However it motivates us more to develop this free open source tool if we know how many people use it in a day-to-day. The transmitted data will be the commit hash of your currently installed version, the timestamp and your IP address. IP addreses are never logged, but we will extract the information of the country as a metric to know our users better.

Now that I read it the text is pretty long ... :)

But as said, this is a first discussion. Happy for feedback!

**Comment by ArneTR** [comment]
Ping for relevant people on this @dan-mm @ribalba @davidkopp @saiteja13427 @shekhar-chauhan

**Comment by ribalba** [comment]
Hey, 

so I really like the idea so that we somehow get an idea of how many people are using us as it is quite weird just to deploy stuff into the void and then get told "Ahhh I love your tool at a conference". I would also include a machine uuid1 (based of the network card) so that we don't get duplicates. We could also include some sort of call to action. Like "Like us on GitHub"

We could be a little cheeky and download a resource while running the install script from one of our servers with a uuid1 as parameter . Like this we would also know how many installs we have. We could also host the docker images our selves to see how often people download stuff. But I am not a big fan of things like this as there is really no real gain for us and privacy invasion is something I am not a big on. Just so it doesn't come up in further discussions. 

Just so that we have it in writing. We have also talked about a data donation function in which runs get submitted to our database. The problem is that saving runs is not cheap in that they use a lot of data and that we don't really gain a lot from the data. It would be sort of ok if we would also know the code but then we don't know how the measurements were made so that data is quite useless. So we decided not to go that route.

**Comment by ArneTR** [comment]
actually the UUID is a good idea, but even for another reason.

The machine_id is currently just an integer. However machines announce themselves on in our cluster on first measurement. It would be way nicer if the machines create a UUID ahead of time that never changes.

This UUID could then be re-used as we anyway need it actually for the cluster to be able to orchestrate automatically.

**Comment by shekhar-chauhan** [comment]
Keep in mind, if devs are to ask users about. 'tracking' the ip for, whatever reasons, say, only to fetch the country of the user, there is a chance, it might go in a negative way. Might be a scenario, where a small paranoia may be developed amongst the users whether you're still going to do it or not. All the big tech have always said they don't log the data, or don't process it in any way, but we all know, its a big fat lie and there have been countless incidents to prove it. 
One way is, we should rather have a poll on the users first about having to have this as a feature or not, and based on that the devs can proceed further.

**Reply by ArneTR** [reply]
Jap, I totally agree. This discussion is exactly for that. We atm do not have any other means of contacting users as the GMT provides no such functionality and there is also no mailing list or such.

THanks for taking the time to write your remarks! We will factor that in.

We plan to leave this discussion open for a while so that users have time to find it.
